packages/eventual-send/src/index.js

```diff
@@ -355,15 +357,36 @@ export function makeHandledPromise() {
   // eslint-disable-next-line prefer-const
   forwardingHandler = {
     get: makeForwarder('get', (o, key) => o[key]),
-    applyMethod: makeForwarder('applyMethod', (o, optKey, args) => {
-      if (optKey === undefined || optKey === null) {
-        return o(...args);
+    applyMethod: makeForwarder('applyMethod', (t, method, args) => {
+      if (method === undefined || method === null) {
+        if (!(t instanceof Function)) {
+          const ftype = typeof t;
+          throw new TypeError(`target is not a function, typeof is ${ftype}`);
+        }
+        return t(...args);
+      }
+      if (!t) {
+        const ftype = typeof t;
+        throw new TypeError(
+          `target cannot contain [${q(method)}], typeof is ${ftype}`,
+        );
+      }
+      if (!(method in t)) {
+        const names = Object.getOwnPropertyNames(t).sort();
+        throw new TypeError(
```

Comment:
I think I used `new Error` when I first started, but since then I've moved to just `Error`. Feel free to stick with your own pre-existing convention instead of copying my confused one :)